### PR TITLE
Fix the policy template details page for Gatekeeper constraints

### PR DESCRIPTION
**Pull request by mprahl**
This fixes the search query, sets the details column to the Gatekeeper constraint's status.violations array, and sets the "Related resources" table based on the audit results.

Relates:
https://issues.redhat.com/browse/ACM-4808

A follow-up PR is required to fix the rest of what's in ACM-4808.

image
image

### CODE REVIEW

**Review by JustinKuli — COMMENTED** [review]
LGTM! Just one tiny thing.

Out of curiosity, do the "view YAML" links work? I'm wondering if the gk audit results will report on objects that were prevented from being created, in which case I think the link would 404 or some other error... to me, that's ok and I don't think we should necessarily handle it here, but I just wanted to bring it up.

File: frontend/src/routes/Governance/policies/policy-details/PolicyTemplateDetails.tsx [hunk 1]
```diff
@@ -102,6 +131,13 @@ export function PolicyTemplateDetails(props: {
     }
   }, [])
 
+  let details
+  if (template?.status?.compliancyDetails) {
+    details = template?.status?.compliancyDetails
+  } else if (template?.status?.violations) {
+    details = template?.status?.violations
+  }
```
[the remark this answers is not in the document]
Comment (mprahl) [reply frontend/src/routes/Governance/policies/policy-details/PolicyTemplateDetails.tsx:139]: Good catch! Thanks.

**Comment by mprahl** [comment]
> LGTM! Just one tiny thing.
> 
> Out of curiosity, do the "view YAML" links work? I'm wondering if the gk audit results will report on objects that were prevented from being created, in which case I think the link would 404 or some other error... to me, that's ok and I don't think we should necessarily handle it here, but I just wanted to bring it up.

Good question. The audit results only include objects that currently exist on the cluster that violate the Gatekeeper constraint.

Any requests that were blocked don't show up in the audit results. Gatekeeper does send events on the cluster for those but we don't capture them in the policy status since it's the Gatekeeper constraint is doing what it's supposed to.

**Review by yiraeChristineKim — APPROVED** [review]

**Pull request merged**